x-pack/plugins/observability/public/pages/overview/old_overview_page.tsx

```diff
@@ -59,6 +59,20 @@ export function OverviewPage({ routeParams }: Props) {
 
   const { hasDataMap, hasAnyData, isAllRequestsComplete } = useHasData();
 
+  const bucketSize = calculateBucketSize({
+    start: absoluteTime.start,
+    end: absoluteTime.end,
+  });
+
+  const bucketSizeValue = useMemo(() => {
+    if (bucketSize?.bucketSize) {
+      return {
+        bucketSize: bucketSize.bucketSize,
+        intervalString: bucketSize.intervalString,
+      };
+    }
+  }, [bucketSize?.bucketSize, bucketSize?.intervalString]);
+
```

Comment:
I'm not sure that `useMemo` is the best option here. `bucketSize` and `bucketSizeValue` are exactly the same value and I don't think it reflects the intentions of what we want to achieve, which could lead to confusion and bugs in the future. 
`bucketSize` should probably be part of the state of the component and we should only update it when `absoluteTime.start` or `absoluteTime.end` change, preventing unnecessary rerenders.

What do you think about something like this:
```suggestion
  const [bucketSize, setBucketSize] = useState(undefined);
  
  useEffect(() => {
     const bucket = calculateBucketSize({
        start: absoluteTime.start,
        end: absoluteTime.end,
      });
      setBucketSize(bucket);
  }, [absoluteTime.start, absoluteTime.end])
  
```

Reply:
i think absolute time start and end changes whenever component re-renders , so can't use those in side effect

Reply:
I see... I wasn't expecting it 🙃